### ISSUE TITLE
Session resumption (caching)

### ISSUE THREAD

**Issue by raciasolvo** [centminmod/centminmod#61]
Hello!

Session resumption (caching) does not work (123.09beta01)

For session resumption (caching) define
```
    ssl_session_cache           shared:SSL:10m;
```
at http{}

ssllabs.com tested.

**Comment by raciasolvo** [comment]
I think that we have to define ssl_session_timeout at http{} too.

**Comment by centminmod** [comment]
are you sure ? what you use to test site with for this ?

seems to be working for me at https://dev.ssllabs.com/ssltest/analyze.html?d=centminmod.com&hideResults=on&latest

![image](https://cloud.githubusercontent.com/assets/5899584/20551816/21636910-b190-11e6-8d6e-d72399e7edc8.png)

**Comment by raciasolvo** [comment]
Do you use your https server as default? My default server is http and have other server name. Read about this problem:
https://forum.nginx.org/read.php?2,152294,152294
http://stackoverflow.com/questions/22732045/session-cache-not-detected-in-nginx

**Comment by raciasolvo** [comment]
Hm... I have A+ (123.09beta01), but centminmod.com has A-rating. I need to see the difference.

**Comment by centminmod** [comment]
https://forum.nginx.org/read.php?2,152294,152401#msg-152401 that shouldn't apply to centmin mod generated https vhost as all have a include file `/usr/local/nginx/conf/ssl_include.conf` which includes

```
ssl_session_cache      shared:SSL:10m;
ssl_session_timeout    60m;
ssl_protocols  TLSv1 TLSv1.1 TLSv1.2; 
```

there is not centmin mod https generated vhost which does not contain the ssl_session_cache directive

**Comment by raciasolvo** [comment]
Yes, we have `include /usr/local/nginx/conf/ssl_include.conf;` at all `server{}` sections by default. You're right.

I understand better the reason why I have the issue. Thank you.
It is possible that one day I commented out the `include` in one of the sections or manually created section or use different `ssl_session_cache`. I do not remember that, when tested the server using ssllabs.com. I did not know about this issue. And I spend some time searching for solution.

Many users do not know about these issue. We can avoid the issue if we use `ssl_session_cache` at the `http{}`.
